src/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,17 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+public interface Applicative<T> extends Apply<T> {
```

Comment:
Agreed Daniel, I think it would be a good addition to the algebra package. I'll keep trying this. I've looked over my code again and it is the same as yours, but I'm still getting the errors on the override method for `ap`. I'm also using IntelliJ 15.0.2 on a Mac with JDK  1.8.0_66. Not sure what the discrepancy is, but I'll keep at it. You've got some good ideas on how to make this work, so I think it will be good once it's done.